### PR TITLE
restart desktop on macos update

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

File: ee/desktop/runner/runner.go [hunk 1]
```diff
@@ -213,6 +225,24 @@ func (r *DesktopUsersProcessesRunner) Execute() error {
...
+			}
+
+			osVersion, err := osversion()
+			if err != nil {
+				level.Error(r.logger).Log("msg", "getting os version", "err", err)
+				continue
+			}
+
+			if osVersion != r.osVersion {
+				r.osVersion = osVersion
```
Comment (zackattack01) [inline ee/desktop/runner/runner.go:243]: would we want some logging here to indicate why the restart happens?

**Review by zackattack01 — DISMISSED** [review]
nice, LGTM! just the one question about logging

File: ee/desktop/runner/runner.go [hunk 2]
```diff
@@ -200,6 +210,8 @@ func (r *DesktopUsersProcessesRunner) Execute() error {
 	defer updateTicker.Stop()
 	menuRefreshTicker := time.NewTicker(r.menuRefreshInterval)
 	defer menuRefreshTicker.Stop()
+	osUpdateCheckTicker := time.NewTicker(5 * time.Second)
```
Comment (RebeccaMahany) [inline ee/desktop/runner/runner.go:213]: Maybe a little less frequently than this? I could see once a minute being tolerable for the end user

**Review by zackattack01 — APPROVED** [review]
🔥

File: ee/desktop/runner/runner.go (hunk 1 above)
Comment (James-Pickett) [reply ee/desktop/runner/runner.go]: added some more logging, put in own func

File: ee/desktop/runner/runner.go (hunk 2 above)
Comment (James-Pickett) [reply ee/desktop/runner/runner.go]: bumped to 1 min

**Review by RebeccaMahany — APPROVED** [review]

**Review by directionless — APPROVED** [review]
neat. I wonder if this crops up on window.

**Pull request merged**